### PR TITLE
Vault-deployment and Grafana-deployment and Nexus-deployment files are created and updated!

### PR DESCRIPTION
**Pull request by Murodbey**
Hello world!
I created terraform deployment file and "nice Readme" file, which you can create vault using terraform apply. I tested and it`s working. Please have a look and approve it. Let me know if there is someth needs to be changed! 

I updated GRAFANA-DEPLOYMENT today, cuz the files what we have on master branch are not gonna work + we were exposing our admin password for grafana in the secrets file. I changed it to variables + ports issue fixed + naming issues fixed + pvc attached and etc. 

I created NEXUS-deployment files today, please have a look. You will need them to create jenkinsfile.groovy.

Please have a look and let us know if u have any questions.
Thank you!

### CODE REVIEW

File: google_vault/vault-deployment.tf [hunk 1]
```diff
@@ -0,0 +1,131 @@
+resource "kubernetes_secret" "vault_secret" {
+  metadata {
+    name      = "vault-secret"
+    namespace = "tools"
+  }
+
+  data {
+    token = "TVJYZ2tTeGZ0c2pqeUlqYkF4Nk9MS0Rmbgo="
```
[the remark this answers is not in the document]
Comment (Murodbey) [reply google_vault/vault-deployment.tf]: done!

File: google_grafana/grafana_deployment.tf [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+              secret_key_ref {
+                name = "grafana-secrets"
+                key  = "username"
+              }
+            }
+          }
+          env {
+            name = "GF_SECURITY_ADMIN_PASSWORD"
+            value_from {
+              secret_key_ref {
```
Comment (fsadykov) [inline google_grafana/grafana_deployment.tf]: I like how you using the secrets in your deployment

File: google_grafana/grafana_deployment.tf [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+  }
+  type = "Opaque"
+}
+
+resource "kubernetes_persistent_volume_claim" "grafana-pvc" {
+  depends_on = ["kubernetes_secret.grafana-secrets"]
+
+  metadata {
+    name      = "grafana-pvc"
+    namespace = "tools"
```
Comment (fsadykov) [inline google_grafana/grafana_deployment.tf]: Suggestion change **tools** to a variable may be in the feature will we change the namespace name

Comment (Murodbey) [reply google_grafana/grafana_deployment.tf]: done!

**Pull request merged**